org.eclipse.xtext/src/org/eclipse/xtext/resource/persistence/SourceLevelURIsAdapter.java

```diff
@@ -53,7 +53,7 @@ protected static SourceLevelURIsAdapter findOrCreateAdapter(ResourceSet resource
 	 */
 	public static void setSourceLevelUrisWithoutCopy(ResourceSet resourceSet, Set<URI> uris) {
```

Comment:
I think so because UnmodifiableSet does not copy the data, it creates a view over it. The idea behing the change is to create the view once instead of each time the data is asked, a small performance optimization.